crates/primitives/src/chain/spec.rs

```diff
@@ -673,6 +673,25 @@ impl ChainSpec {
         }
     }
 
+    /// Get the [BaseFeeParams] for the chain at the given block number
+    pub fn base_fee_params_at_block(&self, block_number: u64) -> BaseFeeParams {
+        match self.base_fee_params {
+            BaseFeeParamsKind::Constant(bf_params) => bf_params,
+            BaseFeeParamsKind::Variable(ForkBaseFeeParams { 0: ref bf_params }) => {
```

Comment:
```suggestion
            BaseFeeParamsKind::Variable(ForkBaseFeeParams(ref bf_params) => {
```

this will require a a `Deref` impl on ForkBaseFeeParams